ocaml/xenforeign/dune

```diff
@@ -0,0 +1,4 @@
+(executable
```

Comment:
Could we make this part of the quicktest? Then it will be run on a host where Xen is available and we can check it doesn't break.  (I think the quicktests can use a CoreOS VM, see quicktest_vm_lifecycle.ml)

Reply:
To run successfully it needs a special xenserver build as well as a windows VM, I don't think we can add this to quicktest as it is. 

The code gets checked in the github CI using `make check`. I think this is OK for the time being, towards the end of the feature when it has stabilized in the main branch we can do this along the upstreaming of the bindings

Reply:
Lets think about the quicktest later. In theory it should be possible to add this device to a Linux VM too. If this depends on any changes in Xen then we may want to avoid running that quicktest by default until the branch is merged.
Shouldn't we do this development on a feature branch though? There are likely to be more revisions to this code, and it is unlikely to interfere with code on master. There is a `feature/vtpm` branch that I've created for this purpose.